integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/hotdeployment/test/SynapseArtifactsHotDeploymentTestCase.java

```diff
@@ -0,0 +1,163 @@
+                "Sequence Deployment failed");
+
+        logViewerClient.clearLogs();
+        FileUtils.touch(new File(sequenceFile));
+        log.info(sequenceFileName + " has been updated and waiting for redeployment");
+        Assert.assertTrue(searchInLogs(logViewerClient, "HotDeploymentTestSequence has been updated from the file"),
+                "Sequence deployment failed on updating file");
+
+        FileManager.deleteFile(sequenceFile);
+        Assert.assertTrue(esbUtils.isSequenceUnDeployed(contextUrls.getBackEndUrl(), sessionCookie, proxyName),
+                "Sequence Undeployment failed");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Endpoint Hot Deployment")
+    public void testEndpointHotDeployment() throws Exception {
+        String endpointFile = SERVER_DEPLOYMENT_DIR + "endpoints" + File.separator + endpointFileName;
+
+        Assert.assertTrue(esbUtils.isEndpointDeployed(contextUrls.getBackEndUrl(), sessionCookie, endpointName),
+                "Endpoint Deployment failed");
+
+        logViewerClient.clearLogs();
+        FileUtils.touch(new File(endpointFile));
+        log.info(endpointFileName + " has been updated and waiting for redeployment");
+        Assert.assertTrue(searchInLogs(logViewerClient, "HotDeploymentTestEndpoint has been updated from the file"),
+                "Endpoint deployment failed on updating file");
+        FileManager.deleteFile(endpointFile);
+        Assert.assertTrue(esbUtils.isEndpointUnDeployed(contextUrls.getBackEndUrl(), sessionCookie, endpointName),
+                "Endpoint Undeployment failed");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void unDeployService() throws Exception {
+        super.cleanup();
+    }
+
+    private void copyArtifactsToDeploymentDirectory() throws IOException {
+        String proxyFile = SOURCE_DIR + proxyFileName;
+        String sequenceFile = SOURCE_DIR + sequenceFileName;
+        String endpointFile = SOURCE_DIR + endpointFileName;
+        FileManager.copyFile(new File(proxyFile),
+                SERVER_DEPLOYMENT_DIR + "proxy-services" + File.separator + proxyFileName);
+        FileManager.copyFile(new File(sequenceFile),
+                SERVER_DEPLOYMENT_DIR + "sequences" + File.separator + sequenceFileName);
+        FileManager.copyFile(new File(endpointFile),
+                SERVER_DEPLOYMENT_DIR + "endpoints" + File.separator + endpointFileName);
+    }
+
+    private boolean searchInLogs(LogViewerClient logViewerClient, String searchString)
+            throws LogViewerLogViewerException, RemoteException, InterruptedException {
+        boolean logFound = false;
+        for (int i = 0; i < 60; i++) {
+            LogEvent[] logEvents = logViewerClient.getAllRemoteSystemLogs();
+            if (logEvents != null) {
+                for (LogEvent logEvent : logEvents) {
+                    if (logEvent == null) {
```

Comment:
Is it possible to logEvent become null?

Reply:
yes. there are some times as i remember, so having a null check for safe

Reply:
Oh ... there are lots of places in our tests that we haven't done that null check